mesonbuild/mesonlib.py

```diff
@@ -1389,13 +1389,13 @@ def __exit__(self, *args):
             msvcrt.locking(self.lockfile.fileno(), msvcrt.LK_UNLCK, 1)
         self.lockfile.close()
 
-def relpath(path, start):
+def relpath(path: str, start: str) -> str:
     # On Windows a relative path can't be evaluated for paths on two different
     # drives (i.e. c:\foo and f:\bar).  The only thing left to do is to use the
     # original absolute path.
     try:
         return os.path.relpath(path, start)
-    except ValueError:
+    except (TypeError, ValueError):
```

Comment:
I don't understand what you do here. When does TypeError exception happens?

Reply:
when `path is None`
I perhaps did not think about the broader context sufficiently. This occurs on Windows.